gitbook-docs/market/overview.md

```diff
@@ -1,63 +1,84 @@
-
-
-* **Sell Order**: A sell order is posted by a proof Generator. Each sell order comprises of following parameters.
-  * **Circuit**: The instruments for which a generator is willing to generate a proof.
-  * **Cost:** Amount the proof generator is willing to accept to create a proof.
-  * **Order Time Out**: Once the order enters the marketplace, this defines how long it should stay in the order book. If it times out before being matched, it is purged. This is a number expressed in minutes.
-  * **Generation Time:** The upper bound by which the proof will be generated once matched.
-
-## Order Matching
-
-Currently, a daemon process matches the orders when the following parameters are met within the order timeout criteria.
-
-* Cost
-* Generation time
-
-## Order Settlement
-
-Order settlement takes place once the proof generator has submitted the proof to the marketplace after generation.
-
+The marketplace consists of the following entities:
+
+* **Proof requester** — an application like a bridge requesting balance or a user
+    interested in a cross-cluster operation or trustless data access.
+* **Proof producer** — an entity that will generate the proofs for the requests/orders
+    made by the proof requester.
+
+## Orders types
+
+* **Buy order**: A buy order is posted by a proof requester.
+    Each buy order comprises the following parameters:
+  * **Statement** (JSON) — the instrument for which this order is placed, like MINA or Solana
+    state-proof circuit, [prepared as a statement](cmd-reference/statement.md/#prepare-statement)
+    with additional parameters like name, URL, etc.
+  * **Cost** (in USD) — the amount that the proof requester is willing to pay for the proof.
+  * **Order time out** (in minutes) — how long an order stays in the order book
+    once it enters the marketplace.
+    The order will be purged if it times out before being matched.
+  * **Generation time** (in minutes, optional) — the maximum time the proof producer
+    has to publish the proof once the order is matched.
+
+* **Sell order**: A sell order is posted by a proof producer.
+    Each sell order has the following parameters:
+  * **Statement** (JSON) — the instruments for which a producer is willing to generate proof.
+  * **Cost** (in USD) — the amount that the proof producer is willing to create a proof for.
+  * **Order time out** (in minutes) — how long an order stays in the order book
+    once it enters the marketplace.
+    The order will be purged if it times out before being matched.
+  * **Generation time** (in minutes, optional) — the maximum time the proof producer
+    has to publish the proof once the order is matched.
+
+## Order matching
+
+Currently, a matching engine matches the orders when the following parameters are met
+within the order timeout criteria:
```

Comment:
Same thing, it doesn't matter what agent performs the matching:

* daemon,
* engine,
* algorithm,
* program,
* code,
* application,
* whatever.

```suggestion
Currently, a matching engine matches the orders when the following parameters are met
within the order timeout criteria:

Buy orders match with sell orders when the following parameters meet:

* cost
* order timeout
* generation time
```

Reply:
A matching engine is an exchange terminology. Recommend we stick to it.
https://www.nasdaq.com/solutions/nasdaq-trading-technology/about-matching-engines

Reply:
Done